base/src/source.rs

```diff
@@ -1,31 +1,74 @@
 use pos::{BytePos, Column, Line, Location};
 
 #[derive(Clone, Debug)]
-pub struct Source<'a> {
-    src: &'a str,
-    /// The starting byte position of each line in `src`
+pub struct Lines {
     lines: Vec<BytePos>,
```

Comment:
Maybe call this `starting_bytes` to reduce stuttering?

Also might make sense to give `Source` and `Lines` doc comments seeing as they are public APIs